content/docs/functions/json_array_elements.md

```diff
@@ -0,0 +1,356 @@
+---
+title: Postgres json_array_elements function
+subtitle: "Expands a JSON array into a set of rows."
+enableTableOfContents: true
+---
+
+
+
+
+Use `json_array_elements` to expand a JSON array into a set of rows, each containing one element of the array. It is a simpler option compared to complex looping logic. It is also more efficient than executing the same operation on the application side by reducing data transfer and processing overhead.
+
+
+
+
+
+
+Function signature:
+
+
+```sql
+json_array_elements(json)
+```
+
+
+## `json_array_elements` example
+
+
+Suppose you have a table with developers info:
+
+
+**developers**
+
+
+| id |  name   |          skills          
+|----|---------|---------------------------
+| 1  | Alice   | ["Java", "Python", "SQL"]
+| 2  | Bob     | ["C++", "JavaScript"]
+| 3  | Charlie | ["HTML", "CSS", "React"]
+
+
+<!--
+```sql
+CREATE TABLE developers (
+ id INT PRIMARY KEY,
+ name TEXT,
+ skills JSON
+);
+
+
+INSERT INTO developers (id, name, skills) VALUES
+ (1, 'Alice', '["Java", "Python", "SQL"]'),
+ (2, 'Bob', '["C++", "JavaScript"]'),
+ (3, 'Charlie', '["HTML", "CSS", "React"]');
+```
+-->
+
+
+Now, let's say you want to extract each individual skill from the skills JSON array. You can use `json_array_elements` for that:
+
+
+```sql
+SELECT id, name, skill
+FROM developers,
+    json_array_elements(skills) AS skill;
+```
+
+
+This returns:
+
+
+| id |  name   |    skill    
+|----|---------|--------------
+| 1  | Alice   | "Java"
+| 1  | Alice   | "Python"
+| 1  | Alice   | "SQL"
+| 2  | Bob     | "C++"
+| 2  | Bob     | "JavaScript"
+| 3  | Charlie | "HTML"
+| 3  | Charlie | "CSS"
+| 3  | Charlie | "React"
+
+
+## Advanced examples
+
+
+### Handling nested data
+
+
+Let's consider a scenario where we have a table storing information about products:
+
+
+**products**
+
+
+| id |  name   |                                details                                
+|----|---------|------------------------------------------------------------------------
+| 1  | T-Shirt | {"sizes": ["S", "M", "L", "XL"], "colors": ["Red", "Blue", "Green"]}
+| 2  | Hoodie  | {"sizes": ["XS", "S", "M", "L", "XL"], "colors": ["Black", "Gray"]}
+| 3  | Dress   | {"sizes": ["S", "M", "L"], "colors": ["Pink", "Purple", "Black"]}
+| 4  | Jeans   | {"sizes": ["28", "30", "32", "34"], "colors": ["Blue", "Black"]}
+| 5  | Jacket  | {"sizes": ["S", "M", "L", "XL"], "colors": ["Black", "Brown", "Navy"]}
+
+
+<!--
```

Comment:
These code comments cause the build to fail. JSX comment tags can be used instead:
{ /* Comment here */ }